libraries/plugins/chain/chain_plugin.cpp

```diff
@@ -328,7 +335,32 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
    if( options.count( "shared-file-scale-rate" ) )
       my->shared_file_scale_rate = options.at( "shared-file-scale-rate" ).as< uint16_t >();
 
+
+   // TODO: temporary solution. DELETE when initminer mining public key is read from get_config
+   my->initMiningPubkeyStr = SOPHIATX_INIT_PUBLIC_KEY_STR;
+#ifdef PRIVATE_NET
+   if (options.count("initminer-mining-pubkey")) {
+      my->initMiningPubkeyStr = options.at( "initminer-mining-pubkey" ).as< std::string >();
+   }
+#endif //PRIVATE_NET
+
+
    auto initial_state = [&] {
+#ifdef PRIVATE_NET
+        if (options.count("initminer-account-pubkey")) {
+           // Creates genesis based on provided initminer account public key
+           genesis_state_type genesis;
+           genesis.genesis_time = time_point_sec::from_iso_string("2018-01-01T08:00:00");
+           genesis.initial_balace = SOPHIATX_INIT_SUPPLY;
+           genesis.initial_public_key = public_key_type(options.at( "initminer-account-pubkey" ).as< std::string >());
```

Comment:
It may crash here in case no initminer-account-pubkey param is provided.

Reply:
Should there be the strict rule: " initminer-account-pubkey must be always provided when starting private net" ? 
in case no initminer-account-pubkey is provided, it tries to read provided genesis-json(i can image you might want to start private net with some initial accounts from custom genesis ? )

Reply:
It shall not start - some proper error message shall be returned. But this code 
```options.at( "initminer-account-pubkey" ).as< std::string >()``` may crash if ```options.at( "initminer-account-pubkey" )``` returns invalid iterator.